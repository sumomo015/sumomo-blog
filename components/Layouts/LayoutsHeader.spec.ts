// @vitest-environment nuxt

import { describe, it, expect } from 'vitest'
import { mountSuspended } from '@nuxt/test-utils/runtime'

import { LayoutsHeader } from '#components'

describe('LayoutsHeader', () => {
  const selectorToogleButton = '[data-testid="toggle-color-scheme-button"]'
  const selectorMenuButton = '[data-testid="toggle-menu-button"]'

  it('スナップショット', async () => {
    const wrapperDark = await mountSuspended(LayoutsHeader, {
      props: { isDarkMode: true },
    })
    expect(wrapperDark.html()).toMatchSnapshot('dark')

    const wrapperLight = await mountSuspended(LayoutsHeader, {
      props: { isDarkMode: false },
    })
    expect(wrapperLight.html()).toMatchSnapshot('light')
  })

  it('カラー切り替えイベントが発火されること', async () => {
    const wrapper = await mountSuspended(LayoutsHeader, {
      props: { isDarkMode: true },
    })
    await wrapper.find(selectorToogleButton).trigger('click')
    expect(wrapper.emitted('toggleColorScheme')).toBeTruthy()
  })

  it('メニュー切り替えイベントが発火されること', async () => {
    const wrapper = await mountSuspended(LayoutsHeader, {
      props: { isDarkMode: true },
    })
    await wrapper.find(selectorMenuButton).trigger('click')
    expect(wrapper.emitted('toggleMenu')?.[0]).toEqual([true])
    await wrapper.find(selectorMenuButton).trigger('click')
    expect(wrapper.emitted('toggleMenu')?.[1]).toEqual([false])
  })
})
