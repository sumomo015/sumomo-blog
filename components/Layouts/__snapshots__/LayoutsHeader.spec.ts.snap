// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LayoutsHeader > スナップショット > dark 1`] = `
"<header class="border-b-1 border-gray-200 dark:border-gray-800 base-bg">
  <div class="flex items-center p-4 pl-6 xl:mx-auto xl:w-screen-xl lg:px-8"><a href="/" class="font-bold base-color-header base-hover-opacity"> Sumomo's Blog </a>
    <div class="flex-grow-1"></div><!-- desktop menu -->
    <nav class="mr-30 hidden gap-x-8 lg:flex" aria-label="Desktop navigation"><a href="/" class="block text-sm font-medium transition-colors hover:text-sky-500 dark:hover:text-sky-400">Home</a><a href="/blog" class="block text-sm font-medium transition-colors hover:text-sky-500 dark:hover:text-sky-400">Blog</a></nav><!-- desktop menu -->
    <!-- mobile menu -->
    <nav class="fixed inset-0 top-16 border-t-1 border-gray-200 bg-white px-4 lg:hidden dark:border-gray-800 dark:bg-gray-900 sm:px-6" aria-label="Mobile navigation" style="display: none;"><a href="/" class="mt-3 flex items-center gap-1.5 text-sm/6 font-medium transition-colors hover:text-sky-500 dark:hover:text-sky-400"><span class="size-5 flex-shrink-0 i-ph-house"></span><span>Home</span></a><a href="/blog" class="mt-3 flex items-center gap-1.5 text-sm/6 font-medium transition-colors hover:text-sky-500 dark:hover:text-sky-400"><span class="size-5 flex-shrink-0 i-ph-newspaper"></span><span>Blog</span></a></nav><!-- mobile menu -->
    <div class="flex gap-x-1.5"><a href="https://github.com/sumomo015" rel="noopener noreferrer" target="_blank" class="size-8 flex items-center justify-center rounded-md base-hover-bg" aria-label="Go to GitHub"><span class="i-simple-icons-github size-5"></span></a><button class="size-8 flex items-center justify-center rounded-md duration-none base-hover-bg" type="button" aria-label="Toggle dark mode" data-testid="toggle-color-scheme-button"><span class="size-5 i-ph-moon"></span></button><button class="size-8 flex items-center justify-center rounded-md lg:hidden base-hover-bg" type="button" aria-label="Toggle navigation menu" data-testid="toggle-menu-button"><span class="size-5 i-ph-list"></span></button></div>
  </div>
</header>"
`;

exports[`LayoutsHeader > スナップショット > light 1`] = `
"<header class="border-b-1 border-gray-200 dark:border-gray-800 base-bg">
  <div class="flex items-center p-4 pl-6 xl:mx-auto xl:w-screen-xl lg:px-8"><a href="/" class="font-bold base-color-header base-hover-opacity"> Sumomo's Blog </a>
    <div class="flex-grow-1"></div><!-- desktop menu -->
    <nav class="mr-30 hidden gap-x-8 lg:flex" aria-label="Desktop navigation"><a href="/" class="block text-sm font-medium transition-colors hover:text-sky-500 dark:hover:text-sky-400">Home</a><a href="/blog" class="block text-sm font-medium transition-colors hover:text-sky-500 dark:hover:text-sky-400">Blog</a></nav><!-- desktop menu -->
    <!-- mobile menu -->
    <nav class="fixed inset-0 top-16 border-t-1 border-gray-200 bg-white px-4 lg:hidden dark:border-gray-800 dark:bg-gray-900 sm:px-6" aria-label="Mobile navigation" style="display: none;"><a href="/" class="mt-3 flex items-center gap-1.5 text-sm/6 font-medium transition-colors hover:text-sky-500 dark:hover:text-sky-400"><span class="size-5 flex-shrink-0 i-ph-house"></span><span>Home</span></a><a href="/blog" class="mt-3 flex items-center gap-1.5 text-sm/6 font-medium transition-colors hover:text-sky-500 dark:hover:text-sky-400"><span class="size-5 flex-shrink-0 i-ph-newspaper"></span><span>Blog</span></a></nav><!-- mobile menu -->
    <div class="flex gap-x-1.5"><a href="https://github.com/sumomo015" rel="noopener noreferrer" target="_blank" class="size-8 flex items-center justify-center rounded-md base-hover-bg" aria-label="Go to GitHub"><span class="i-simple-icons-github size-5"></span></a><button class="size-8 flex items-center justify-center rounded-md duration-none base-hover-bg" type="button" aria-label="Toggle dark mode" data-testid="toggle-color-scheme-button"><span class="size-5 i-ph-sun"></span></button><button class="size-8 flex items-center justify-center rounded-md lg:hidden base-hover-bg" type="button" aria-label="Toggle navigation menu" data-testid="toggle-menu-button"><span class="size-5 i-ph-list"></span></button></div>
  </div>
</header>"
`;
